scripts/lint_jjb.py

```diff
@@ -155,6 +168,149 @@ def check_retention(job, in_file):
     return return_value
 
 
+# Ensure all jobs have a valid timed triggers entry
+# Structure:
+#   job:
+#     triggers:
+#      - timed: "@daily"
+def check_timed_trigger_value(trigger):
+
+    # Non-standard CRON values allowed by jenkins
+    # Reference:
+    # https://en.wikipedia.org/wiki/Cron#Nonstandard_predefined_scheduling_definitions
+    # http://www.scmgalaxy.com/tutorials/setting-up-the-cron-jobs-in-jenkins-using-build-periodically-scheduling-the-jenins-job/
+    allowed_non_standard_values = [
+        "@yearly", "@annually", "@monthly", "@weekly", "@daily",
+        "@midnight", "@hourly"
+    ]
+
+    # If the trigger provided is a NoneType,
+    # then the job has no value, so pass it.
+    if trigger is None:
+        return_value = 0
+    # If the trigger provided is not a string,
+    # then it is not valid.
+    elif not isinstance(trigger, str):
+        return_value = 1
+    # If the CRON macro is used, we do not need
+    # to validate it.
+    elif trigger == "{CRON}":
+        return_value = 0
+    # If non-standard predefined schedule definitions
+    # are used, pass them.
+    elif trigger in allowed_non_standard_values:
+        return_value = 0
+    else:
+        # Jenkins supports the use of hashes, denoted by the symbol H,
+        # in the trigger to allow the load to be spread on the system.
+        # The croniter library does not know how to validate them, so
+        # we replace them.
+        # Valid inputs and what they should output after replacement:
+        # 1. "H * * * *" > "* * * * *"
+        # 2. "H(0-29)/10 * * * *" > "0-29/10 * * * *"
+        # 3. "H/10 H(0-11) * H *" > "*/10 0-11 * * *"
+
+        # handle the H(..) pattern first
+        _trigger = re.sub(r'H\((\d-\d+)\)', r'\1', trigger)
```

Comment:
I think the range part `\d-\d+` should probably be changed, the first part of the range could be double digits as well.

Reply:
Very good point - odd that it didn't error out though because I used the test data mentioned in the lines above to validate things. Are you able to figure out a bit of test data that breaks with this test?

Reply:
"H/10 H(10-11) * H *"

Reply:
Of course. Silly me. Updating it now.